src/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine.php

```diff
@@ -56,57 +48,71 @@ class SearchEngine implements ProcessInterface
      */
     private $searchEngineConfig;
 
+    /**
+     * @var ElasticSuite
+     */
+    private $elasticSuite;
+
     /**
      * @param LoggerInterface $logger
      * @param EnvWriter $envWriter
-     * @param EnvReader $envReader
      * @param SharedWriter $sharedWriter
-     * @param SharedReader $sharedReader
      * @param MagentoVersion $version
      * @param SearchEngineConfig $searchEngineConfig
+     * @param ElasticSuite $elasticSuite
      */
     public function __construct(
         LoggerInterface $logger,
         EnvWriter $envWriter,
-        EnvReader $envReader,
         SharedWriter $sharedWriter,
-        SharedReader $sharedReader,
         MagentoVersion $version,
-        SearchEngineConfig $searchEngineConfig
+        SearchEngineConfig $searchEngineConfig,
+        ElasticSuite $elasticSuite
     ) {
         $this->logger = $logger;
         $this->envWriter = $envWriter;
-        $this->envReader = $envReader;
         $this->sharedWriter = $sharedWriter;
-        $this->sharedReader = $sharedReader;
         $this->magentoVersion = $version;
         $this->searchEngineConfig = $searchEngineConfig;
+        $this->elasticSuite = $elasticSuite;
     }
 
     /**
      * Executes the process.
      *
      * @return void
+     * @throws ProcessException
      */
     public function execute()
     {
-        $this->logger->info('Updating search engine configuration.');
-
         $searchConfig = $this->searchEngineConfig->get();
 
+        $this->logger->info('Updating search engine configuration.');
         $this->logger->info('Set search engine to: ' . $searchConfig['engine']);
 
-        // 2.1.x requires search config to be written to the shared config file: MAGECLOUD-1317
-        if (!$this->magentoVersion->isGreaterOrEqual('2.2')) {
-            $config = $this->sharedReader->read();
-            $config['system']['default']['catalog']['search'] = $searchConfig;
-            $this->sharedWriter->create($config);
+        try {
```

Comment:
It's not a good approach to have more than one try/catch block in one method.
You can use multiple catch block to handle different Exception.